File: rgctl-tests/ecommerce-python/app/cfg_features.py
"""CFG feature probes for rgctl expected-facts (Python lowering coverage)."""


def cfg_short_circuit(a: bool, b: bool) -> int:
    if a and b:
        return 1
    return 0


def cfg_for_loop(xs: list[int]) -> int:
    total = 0
    for v in xs:
        total += v
    return total


def cfg_try_except() -> int:
    try:
        raise ValueError("boom")
    except ValueError:
        return 1

File: rgctl-tests/ecommerce-typescript/src/cfg_features.ts
/**
 * CFG feature probes for rgctl expected-facts (TypeScript lowering coverage).
 */

export function cfgShortCircuit(a: boolean, b: boolean): number {
  if (a && b) {
    return 1;
  }
  return 0;
}

export function cfgForLoop(xs: number[]): number {
  let total = 0;
  for (let i = 0; i < xs.length; i++) {
    total += xs[i];
  }
  return total;
}

export function cfgTryCatch(): number {
  try {
    throw new Error("boom");
  } catch {
    return 1;
  }
}

export function cfgSwitch(x: number): number {
  switch (x) {
    case 1:
      return 10;
    case 2:
      return 20;
    default:
      return 0;
  }
}
